hyperspy/io_plugins/nexus.py

```diff
@@ -0,0 +1,1130 @@
+    nexus_data_paths,hdf_dataset_paths = \
+        _find_data(fin,search_keys=search_keys)
+    if verbose:
+        if nexus_data_paths:
+            print("NXdata found")
+            for nxd in nexus_data_paths:
+                print(nxd)
+        else:
+            print("NXdata not found")
+        if hdf_dataset_paths:    
+            print("Unique HDF datasets found")
+            for hdfd in hdf_dataset_paths:
+                print(hdfd,fin[hdfd].shape)
+        else:
+            print("No HDF datasets not found or data is captured by NXdata")
+    fin.close() 
+    return nexus_data_paths,hdf_dataset_paths
+
+def _write_signal(signal, nxgroup, signal_name, **kwds):   
+    """ Store the signal data as an NXdata dataset 
+    
+    Parameters
+    ----------
+    signal : Hyperspy signal
+    nxgroup : HDF group 
+        Entry at which to save signal data   
+    signal_name : str    
+        Name  under which to store the signal entry in the file
+            
+    
+    """
+    smd = signal.metadata.Signal
+    if signal.axes_manager.signal_dimension == 1:
+        smd.record_by = "spectrum"
+    elif signal.axes_manager.signal_dimension == 2:
+        smd.record_by = "image"
+    else:
+        smd.record_by = ""
+
+    nxdata = nxgroup.require_group(signal_name)
+    nxdata.attrs["NX_class"] =_parse_to_file("NXdata")  
+    nxdata.attrs["signal"] = _parse_to_file("data")
+    datastr = _parse_to_file("data")
+    overwrite_dataset(nxdata, signal.data, datastr, chunks=None, **kwds)
+    axis_names=[_parse_to_file(".")]*len(signal.axes_manager.shape)
+    for i,axis in enumerate(signal.axes_manager._axes):
+        if axis.name != t.Undefined:
+            axname = _parse_to_file(axis.name)
+            axindex = [axis.index_in_array]
+            indices = _parse_to_file(axis.name + "_indices")
+            nxdata.attrs[indices] = _parse_to_file(axindex)
+            #navigate = _parse_to_file(axis.name + "_navigation")
+            #nxdata.attrs[navigate] = _parse_to_file(axis.navigate)
+            dset = nxdata.require_dataset(axname,data=axis.axis,shape=axis.axis.shape,
+                                   dtype=axis.axis.dtype)
+            dset.attrs["navigation"] = _parse_to_file(axis.navigate)
+            axis_names[axis.index_in_array]=axname
+
+    nxdata.attrs["axes"] = _parse_to_file(axis_names)
+    return nxdata
+
+def file_writer(filename,
+                signals,
+                save_original_metadata=True,
+                *args, **kwds):
+    """ Write the signal and metadata as a nexus file
+    This will save the signal in NXdata format in the file.
+    As the form of the metadata can vary and is not validated it will 
+    be stored as an NXcollection (an unvalidated collection)
+
+    Parameters
+    ----------
+    filename : str    
+        Path of the file to write
+    signals : signal or list of signals
+    store_original_metadata : bool , default : False
```

Comment:
There needs to be a description with a tabulation for sphinx to render the docstring correctly, something along the lines of will do:
```suggestion
    signals : signal or list of signals
        Signal(s) to be written
    small_metadata_only : bool , default : False
```